tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -4747,86 +4733,132 @@ TEST_F(OpConverterTest, ConvertConv2DBackpropInput) {
                  /*data_format=*/"NHWC",
                  /*dilations=*/{1, 1, 1, 1},
-                 /*expected_output_dims=*/{7, 1, 1},
+                 /*expected_output_dims=*/{1, 7, 1, 1},
                  /*expected_output=*/{0, 0, -1, 1, -2, 2, 0}},
+      TestParams{/*input_dims=*/{1, 1, 2, 2},
+                 /*input=*/{0, 1, 2, 3},
+                 /*filter_dims=*/{1, 2, 1, 1},
+                 /*filter=*/{-1, 1},
+                 /*strides=*/{1, 1, 1, 2},
+                 /*padding=*/"EXPLICIT",
+                 /*data_format=*/"NCHW",
+                 /*dilations=*/{1, 1, 1, 1},
+                 /*expected_output_dims=*/{1, 1, 2, 4},
+                 /*expected_output=*/{0, 0, -1, 1, -2, 2, -3, 3},
+                 errors::Unimplemented("EXPLICIT padding type not "
+                                       "implemented, only VALID and SAME are"
+                                       " supported")},
+      // Dilation + Conv2DBackpropInput, should fail.
+      TestParams{/*input_dims=*/{1, 1, 2, 2},
+                 /*input=*/{0, 1, 2, 3},
+                 /*filter_dims=*/{1, 2, 1, 1},
+                 /*filter=*/{-1, 1},
+                 /*strides=*/{1, 1, 1, 1},
+                 /*padding=*/"SAME",
+                 /*data_format=*/"NCHW",
+                 /*dilations=*/{1, 1, 1, 2},
+                 {1, 1, 2, 2},
+                 {},
+                 errors::Unimplemented("Dilation with Conv2DBackpropInput "
+                                       "(conv2d_transpose) is not supported, "
+                                       "at my_conv2d_backprop_input")},
   };
-
-  for (int i = 0; i < ok_params.size(); i++) {
+  if (trt_mode_ == TrtTestMode::kDynamicShape) {
+    params.push_back(TestParams{
+        /*input_dims=*/{1, 1, 2, 2},
+        /*input=*/{0, 1, 2, 3},
+        /*filter_dims=*/{1, 2, 1, 1},
+        /*filter=*/{-1, 1},
+        /*strides=*/{1, 1, 1, 2},
+        /*padding=*/"SAME",
+        /*data_format=*/"NCHW",
+        /*dilations=*/{1, 1, 1, 1},
+        /*expected_output_dims=*/{1, 1, 2, 4},
+        /*expected_output=*/{0, 0, -1, 1, -2, 2, -3, 3},
+        errors::InvalidArgument(
+            "Channel dimension must be static, at my_conv2d_backprop_input"),
+        1});
+  }
+  for (auto p : params) {
     for (int input_sizes_length : {2, 4}) {
       Reset();
       NodeDef node_def = get_conv2d_backprop_input_nodedef(
-          ok_params[i].strides, ok_params[i].padding, ok_params[i].data_format,
-          ok_params[i].dilations);
-      AddTestTensor("input", ok_params[i].input_dims);
-      AddTestWeights<float>("weights", ok_params[i].filter_dims,
-                            ok_params[i].filter);
+          tf_type_, p.strides, p.padding, p.data_format, p.dilations);
+
+      std::vector<int> partial_input_shape;
+      if (trt_mode_ == TrtTestMode::kDynamicShape && !p.unknown_channel) {
+        // The channel dim cannot have unknown size, fix that.
```

Comment:
Sorry, it is not clear to me what this comment intends to explain.
Can you provide more details?
We don't execute the if-stmt body if p.unknown_channel is true. That is all I can tell here...

Reply:
I have updated the explanation.